scylla-cdc-lib/src/test/java/com/scylladb/cdc/lib/AlterTableIT.java

```diff
@@ -167,4 +178,193 @@ private synchronized void printDetails(RawChange change) {
+                   .addTable(new TableName(keyspace, table))
+                   .withConsumer(changeConsumer)
+                   .withQueryTimeWindowSizeMs(15 * 1000)
+                   .withConfidenceWindowSizeMs(confidenceWindowSeconds * 1000)
+                   .withWorkersCount(1)
+                   .withQueryOptionsFetchSize(1)
+                   .build()) {
+        ReentrantReadWriteLock nextPageLock = null;
+        try {
+          // Access the LocalTransport field in CDCConsumer
+          java.lang.reflect.Field transportField = CDCConsumer.class.getDeclaredField("transport");
+          transportField.setAccessible(true);
+          LocalTransport transport = (LocalTransport) transportField.get(consumer);
+
+          // Access the WorkerConfiguration.Builder field in LocalTransport
+          java.lang.reflect.Field builderField = LocalTransport.class.getDeclaredField("workerConfigurationBuilder");
+          builderField.setAccessible(true);
+          WorkerConfiguration.Builder builder = (WorkerConfiguration.Builder) builderField.get(transport);
+
+          // Access the session field in CDCConsumer
+          java.lang.reflect.Field sessionField = CDCConsumer.class.getDeclaredField("session");
+          sessionField.setAccessible(true);
+          Driver3Session driver3Session = (Driver3Session) sessionField.get(consumer);
+
+          // Create MockDriver3WorkerCQL and replace the cql field in the builder
+          MockDriver3WorkerCQL mockWorkerCQL = new MockDriver3WorkerCQL(driver3Session);
+          // Grab the lock to control it later on
+          nextPageLock = mockWorkerCQL.nextPageLock;
+          java.lang.reflect.Field cqlField = WorkerConfiguration.Builder.class.getDeclaredField("cql");
+          cqlField.setAccessible(true);
+          cqlField.set(builder, mockWorkerCQL);
+        } catch (Exception e) {
+          throw new RuntimeException("Failed to replace Driver3WorkerCQL with MockDriver3WorkerCQL", e);
+        }
+        Uninterruptibles.sleepUninterruptibly(confidenceWindowSeconds, TimeUnit.SECONDS);
+        // Let some records be generated first
+        Awaitility.await().atMost(5, TimeUnit.SECONDS).until(() -> {
+          if (counter.get() > 15) {
+            return true;
+          }
+          return false;
+        });
+        nextPageLock.writeLock().lock();
+        consumer.start();
+        // Wait until all threads need to iterate to the next page
+        // We consider the consumer to be blocked by the nextPageLock when the lock has queued threads
+        // and consumer has not consumed any change for the last 10 seconds
+        ReentrantReadWriteLock finalNextPageLock = nextPageLock;
+        Awaitility.await().atMost(60, TimeUnit.SECONDS).until(() -> finalNextPageLock.hasQueuedThreads() && (System.currentTimeMillis() - lastConsumedTime.get() > 10000));
+        ResultSet rs = session.execute(String.format("ALTER TABLE %s.%s " + "ADD column4 int;", keyspace, table));
```

Comment:
Can we also have tests on:
1. Updating UDT
2. Removing column
3. Changing column type (by removing and adding column with the same name, but different type)

It would be great if you can generalize the test

Reply:
All tests listed here were implemented